ara/clients/http.py

```diff
@@ -85,12 +85,12 @@ def _request(self, method, url, **kwargs):
             response = func(url, **kwargs)
 
         if response.status_code >= 500:
-            self.log.error("Failed to {method} on {url}: {content}".format(method=method, url=url, content=kwargs))
+            self.log.error(f"Failed to {method} on {url}: {kwargs}")
 
-        self.log.debug("HTTP {status}: {method} on {url}".format(status=response.status_code, method=method, url=url))
+        self.log.debug(f"HTTP {response.status_code}: {method} on {url}")
 
         if response.status_code not in [200, 201, 204]:
-            self.log.error("Failed to {method} on {url}: {content}".format(method=method, url=url, content=kwargs))
+            self.log.error(f"Failed to {method} on {url}: {kwargs}")
```

Comment:
Yay, f-strings :tada:

Reply:
f-strings shouldn't be used in log module, instead the paramaters should be passed as arguments.

E.g.

``` python
log.error("Hello %s", world)
```

While f-strings will always be evaluated, this will only be evaluated if the log level is above the loggers log level.

See https://docs.python.org/3/howto/logging.html#logging-variable-data 
or https://blog.pilosus.org/posts/2020/01/24/python-f-strings-in-logging/

By using pylint you would also get a warning (https://pylint.pycqa.org/en/latest/user_guide/messages/warning/logging-fstring-interpolation.html)

PS: also the old code with `{}.format` should not be used in log module. But if we change it now, we can make it directly perfect.

Reply:
@hille721 Thanks for the information. I did some google'ing and found this one too. 

https://docs.python.org/3/howto/logging.html#optimization

I will make the changes to use % for log messages, and also run pylint

Reply:
I went ahead and opened a bug report on pyupgrade to address this. 

Ref https://github.com/asottile/pyupgrade/issues/702

Reply:
Pyupgrade developer immediately closed by issue stating "I was wrong and should look closer".

Reply:
maybe because it is not really related to pyupgrade. The behavior of the logging module didn't changed.
I guess pyupgrade is simple replacing str.format strings with f-strings. 
But yeah his answer is really harsh.

Reply:
indeed -- and if you searched the tracker before making an issue you would have found a duplicate -- https://github.com/asottile/pyupgrade/issues?q=is%3Aissue+logging+is%3Aclosed -- garbage in garbage out -- technically pyupgrade _improved_ your performance here but it was incorrect to begin with

Reply:
@asottile there was no need to be mean about it but understood, thanks.

Reply:
not sure where you're interpreting "mean" -- the issue was invalid, a duplicate, no code sample, no version information, and therefore closed